web/src/org/purefn/kurosawa/web/prometheus.clj

```diff
@@ -213,23 +215,25 @@
                  label-fn (constantly {})}
             :as options}]
   (fn [request]
-    (run-instrumented {:path-fn  path-fn
-                       :label-fn label-fn
-                       :registry (:registry (app/service request :prometheus))
-                       :handler  handler}
-                      request)))
+    (let [prometheus (app/service request :prometheus)]
+      (run-instrumented {:path-fn  path-fn
+                         :label-fn label-fn
+                         :registry (when prometheus (:registry prometheus))
```

Comment:
This `when` will pass `nil` `prometheus` is nil, same as before.  How about only executing `run-instrumented` when `prometheus` is truthy?

If we've come this far though (used `wrap-stateful-metrics` but `(app/service req :prometheus)` returns `nil`) it might be worth logging out to let us know we've defied expectations.  Something like this maybe?

```clojure
(if-let [registry (:registry (app/service request :prometheus))]
      (run-instrumented {:path-fn  path-fn
                         :label-fn label-fn
                         :registry registry
                         :handler  handler}
                        request)
      (do (log/warn "...")
          (handler request)))
```

I think the rest of the lower level `when` in this PR could then be removed.